api/types.proto

```diff
@@ -458,6 +458,12 @@ enum TaskState {
 	SHUTDOWN = 640 [(gogoproto.enumvalue_customname)="TaskStateShutdown"]; // orchestrator requested shutdown
 	FAILED = 704 [(gogoproto.enumvalue_customname)="TaskStateFailed"]; // task execution failed with error
 	REJECTED = 768 [(gogoproto.enumvalue_customname)="TaskStateRejected"]; // task could not be executed here.
+	// The main purpose of the REMOVE state is to correctly handle service deletions
+	// and scale downs. This allows us to keep track of tasks that have been marked
+	// for deletion, but can't yet be removed because the agent is in the process of
+	// shutting them down. Once the agent has shut down tasks with desired state
+	// REMOVE, the task reaper is responsible for removing them.
+	REMOVE = 800 [(gogoproto.enumvalue_customname)="TaskStateRemove"];
```

Comment:
`832` for consistency with the others?

Reply:
i think you missed the `ORPHANED = 832` below, which makes this correct at halfway between `REJECTED` and `ORPHANED`.

Reply:
`// Remove ...`